src/Controls/src/Core/VisualElement/VisualElement.cs

```diff
@@ -1395,50 +1397,44 @@ internal virtual void InvalidateMeasureInternal(InvalidationEventArgs eventArgs)
 					break;
 			}
 
-			FireMeasureChanged(eventArgs);
+			InvokeMeasureInvalidated(trigger);
+#pragma warning disable CS0618 // Type or member is obsolete
+			(Parent as VisualElement)?.OnChildMeasureInvalidated(this, trigger);
+#pragma warning restore CS0618 // Type or member is obsolete
 		}
 
-		private protected void FireMeasureChanged(InvalidationTrigger trigger, int depth)
+		private protected void InvokeMeasureInvalidated(InvalidationTrigger trigger)
 		{
-			FireMeasureChanged(new InvalidationEventArgs(trigger, depth));
+			MeasureInvalidated?.Invoke(this, new InvalidationEventArgs(trigger));
 		}
 
+		/// <summary>
+		/// A flag that determines whether the measure invalidated event should be propagated up the visual tree.
+		/// </summary>
+		internal static bool IsMeasureInvalidatedPropagationEnabled { get; set /* for testing purpose */; } =
+			AppContext.TryGetSwitch("Microsoft.Maui.RuntimeFeature.PropagateMeasureInvalidated", out var enabled) && enabled;
```

Comment:
We should default this to `true`, and if not possible, just revert the logic.
The reason is: some platform renderers, like the shell footer on iOS still rely on `MeasureInvalidated` event to detect measure changes.
Until those will be refactored, we have to propagate the measure invalidation.